content/en/docs/instrumentation/swift/automatic.md

```diff
@@ -0,0 +1,125 @@
+
+`shouldInjectTracingHeaders: ((URLRequest) -> (Bool)?)?`: Allows filtering which
+requests you want to inject headers to follow the trace, true by default. You
+must also return true if you want to inject custom headers.
+
+`injectCustomHeaders: ((inout URLRequest, Span?) -> Void)?`: Implement this
+callback to inject custom headers or modify the request in any other way
+
+`nameSpan: ((URLRequest) -> (String)?)?` - Modifies the name for the given
+request instead of stantard Opentelemetry name
+
+`createdRequest: ((URLRequest, Span) -> Void)?` - Called after request is
+created, it allows to add extra information to the Span
+
+`receivedResponse: ((URLResponse, DataOrFile?, Span) -> Void)?`- Called after
+response is received, it allows to add extra information to the Span
+
+`receivedError: ((Error, DataOrFile?, HTTPStatus, Span) -> Void)?` - Called
+after an error is received, it allows to add extra information to the Span
+
+### Details
+
+`NSURLSession` instrumentation also provides additional attributes providing
+details about the network state of the device at the time of network requests.
+
+| Attribute                     | Value example                 | Description                                                                                 |
+| ----------------------------- | ----------------------------- | ------------------------------------------------------------------------------------------- |
+| `net.host.connection.type`    | `wifi`, `cell`, `unavailable` | The type of connection utilized by the device at the time of the request.                   |
+| `net.host.connection.subtype` | `EDGE` `LTE`, etc             | They type of cellular connection. Only populated if the connection type is `cell`           |
+| `net.host.carrier.name`       | `T-Mobile`, `Verison`, etc    | The cellular carrier name. Only populated for cellular connection types.                    |
+| `net.host.carrier.icc`        | `DE`                          | The ISO 3166-1 alpha-2 2-character country code associated with the mobile carrier network. |
+| `net.host.carrier.mcc`        | `310`                         | Mobile Country Code                                                                         |
+| `net.host.carrier.mnc`        | `001`                         | Mobile network code                                                                         |
+
+## `SignpostIntegration`
+
+This package creates `os_signpost` `begin` and `end` calls when spans are
+started or ended. It allows automatic integration of applications instrumented
+with OpenTelemetry to show their spans in a profiling app like `Instruments`. It
+also exports the `OSLog` it uses for posting so the user can add extra signpost
+events. This functionality is shown in `Simple Exporter` example
+
+### Usage
+
+Just add SignpostIntegration as any other Span Processor:
+
+```
+OpenTelemetry.instance.tracerProvider.addSpanProcessor(SignPostIntegration())`
```

Comment:
Where in the code would this line get added? If there's a convention based on file name or a particular class where initialization typically happens then that would be great to include in the sample.